six/modules/c++/cphd/include/cphd/PVP.h

```diff
@@ -435,6 +501,27 @@ struct Pvp
      *  Marks filled bytes
      */
     std::vector<bool> mParamLocations;
+
+    /*
+     * Set default size and format for each parameter
+     */
+    void setDefaultValues(size_t size, const std::string& format, PVPType& param);
+
+    /*
+     *  Optional parameter flags
+     */
+    bool mAmpSF;

```

Comment:
Can you make it work without this? I'd like checking for optional parameters to be consistent across the library, if possible.

Reply:
CPHD03::VBM.h does this too so I was trying to be consistent with that, although CPHD03 does only has 3 optional parameters. I can move the flags to PvpBlock which is the CHPD10 equivalent of VBM.h if that's slightly better.
Or did you have some other case of optional parameter handling elsewhere in mind?

Reply:
Fair, that makes sense.